src/extractor/ConfigurationExtractor.ts

```diff
@@ -0,0 +1,113 @@
+      resources,
+      'status'
+    ) as ImplementationGuideStatus;
+    const version = ConfigurationExtractor.inferString(resources, 'version');
+    if (status) {
+      config.config.status = status;
+    }
+    if (version) {
+      config.config.version = version;
+    }
+    return config;
+  }
+
+  static inferCanonical(resources: Package): string {
+    // url may be present as a CaretValueRule, path = '', caret path = 'url'
+    const potentialCanonicals: string[] = [];
+    [...resources.profiles, ...resources.extensions].forEach(resource => {
+      const urlRule = resource.rules.find(rule => {
+        return (
+          rule instanceof ExportableCaretValueRule && rule.path === '' && rule.caretPath === 'url'
+        );
+      }) as ExportableCaretValueRule;
+      if (urlRule) {
+        potentialCanonicals.push(
+          urlRule.value.toString().replace(/\/StructureDefinition\/[^/]+$/, '')
+        );
+      }
+    });
+    resources.valueSets.forEach(valueSet => {
+      const urlRule = valueSet.rules.find(rule => {
+        return (
+          rule instanceof ExportableCaretValueRule && rule.path === '' && rule.caretPath === 'url'
+        );
+      }) as ExportableCaretValueRule;
+      if (urlRule) {
+        potentialCanonicals.push(urlRule.value.toString().replace(/\/ValueSet\/[^/]+$/, ''));
+      }
+    });
+    resources.codeSystems.forEach(codeSystem => {
+      const urlRule = codeSystem.rules.find(rule => {
+        return (
+          rule instanceof ExportableCaretValueRule && rule.path === '' && rule.caretPath === 'url'
+        );
+      }) as ExportableCaretValueRule;
+      if (urlRule) {
+        potentialCanonicals.push(urlRule.value.toString().replace(/\/CodeSystem\/[^/]+$/, ''));
+      }
+    });
+
+    return ConfigurationExtractor.getStringMode(potentialCanonicals);
+  }
+
+  static inferString(resources: Package, caretPath: string): string {
+    const potentialStrings: string[] = [];
+    [
+      ...resources.profiles,
+      ...resources.extensions,
+      ...resources.valueSets,
+      ...resources.codeSystems
+    ].forEach(resource => {
+      const caretRule = (resource.rules as ExportableRule[]).find(rule => {
+        return (
+          rule instanceof ExportableCaretValueRule &&
+          rule.path === '' &&
+          rule.caretPath === caretPath
+        );
+      }) as ExportableCaretValueRule;
+      if (caretRule) {
+        // if we find a FshCode, we only want the "code" part
+        if (caretRule.value instanceof fshtypes.FshCode) {
+          potentialStrings.push(caretRule.value.code);
+        } else {
+          potentialStrings.push(caretRule.value.toString());
+        }
+      }
+    });
+    return ConfigurationExtractor.getStringMode(potentialStrings);
+  }
+
+  private static getStringMode(potentials: string[]): string {
+    if (potentials.length > 0) {
+      return maxBy(toPairs(countBy(potentials)), pair => pair[1])[0];
```

Comment:
This made my head hurt.  It's right.  But it made my head hurt.  I don't know if the following is necessary or if it would survive `prettier`'s weird code-formatting algorithms, but this helped me understand it:
```ts
      return maxBy(
        toPairs(
          countBy(
            potentials        // e.g., ['foo', 'bar', 'baz', 'foo', 'bar', 'bar']
          )                   // e.g., { 'foo': 2, 'bar': 3, 'baz': 1 }
        ),                    // e.g., [ ['foo', 2], ['bar': 3], ['baz': 1] ]
        pair => pair[1]       // e.g., ['foo', 2] => 2; ['bar': 3] => 3; ['baz': 1] => 1
      )[0];                   // e.g., ['bar', 3][0] -----> 'bar'
```

Reply:
prettier makes it end up looking like this:
```typescript
    if (potentials.length > 0) {
      return maxBy(
        toPairs(
          countBy(
            // this is a little dense. here's an example:
            potentials // ['a', 'b', 'b', 'a', 'c', 'b']
          ) // {'a': 2, 'b': 3, 'c': 1}
        ), // [['a', 2], ['b', 3], ['c', 1]]
        pair => pair[1] // ['a', 2] => 2, ['b', 3] => 3, ['c', 1] => 1
      )[0]; // ['b', 3][0] ---> 'b'
    }
```
So if that seems helpful, I'll include it.

Reply:
I don't know.  Not having control over the spacing makes it a little less helpful.  Let's get a 3rd opinion.  @ngfreiter , @jafeltra , @julianxcarter -- do you think it's better just to leave this line as-is (`return maxBy(toPairs(countBy(potentials)), pair => pair[1])[0];`) or to go with the annotated post-prettier format @mint-thompson posted above?

Reply:
Can we leave it one line with the annotation above it? I think the annotations are nice, but post-prettier they make things more confusing than just no annotations at all

Reply:
Or even maybe just a comment indicating what this line of code does at a general level, if you don't want to include the full step by step process.

Reply:
@ngfreiter -- do you mean:
```ts
      // return maxBy(
      //   toPairs(
      //     countBy(
      //       potentials    // e.g., ['foo', 'bar', 'baz', 'foo', 'bar', 'bar']
      //     )               // e.g., { 'foo': 2, 'bar': 3, 'baz': 1 }
      //   ),                // e.g., [ ['foo', 2], ['bar': 3], ['baz': 1] ]
      //   pair => pair[1]   // e.g., ['foo', 2] => 2; ['bar': 3] => 3; ['baz': 1] => 1
      // )[0];               // e.g., ['bar', 3][0] -----> 'bar'
      return maxBy(toPairs(countBy(potentials)), pair => pair[1])[0];
```

Reply:
Or:
```ts
      // Example results after each function
      //   potentials: ['foo', 'bar', 'baz', 'foo', 'bar', 'bar']
      //   countBy:    { 'foo': 2, 'bar': 3, 'baz': 1 }
      //   toPairs:    [ ['foo', 2], ['bar': 3], ['baz': 1] ]
      //   maxBy:      ['bar', 3]
      //   maxyBy[0]:  'bar'
      return maxBy(toPairs(countBy(potentials)), pair => pair[1])[0];
```

Reply:
Yep that's pretty much what I meant. But I meant I'd also be fine with something like:
```
      // Find the most frequently occurring potential
      return maxBy(toPairs(countBy(potentials)), pair => pair[1])[0];
```

Reply:
Oh when I posted this comment I hadn't seen your "Or". Your "Or" is exactly what I meant.

Reply:
I feel like `// Find the most frequently occurring potential` doesn't add much since the function is called `getStringMode` -- which implies finding the most frequent value already.  @mint-thompson - how do you feel about my suggestion in the "Or:" above?

Reply:
I like it! I'll add something in that format.